Clarify multiverse documentation

ffranr opened the pull request:
I think this change is correct. Opened this PR to check my understanding.

Discussed this a bit with @jharveyb (thanks again!).

Roasbeef commented on bip-tap-universe.mediawiki:
```diff
@@ -253,8 +253,11 @@ A Multiverse has the following structure:
 
-A Multiverse therefore commits to the set of known genesis IDs, and at a second
-level the set of complete Universe trees for each watched asset.
+A Multiverse therefore commits to the set of known genesis IDs, and at a second level the set of complete Universe trees
```
Hmm, isn't this clear with the above? In that the leaves of the upper tree are the roots of the lower tree?

ffranr replied:
In the latest commits, I've removed the additional comments.

ffranr committed:
bip-tap: remove asset_group_sum concatenation in upper tree value field

This concatenation was unnecessary because this value is present in the
upper tree `sum_value` field.

Roasbeef commented on bip-tap-universe.mediawiki:
```diff
@@ -246,7 +246,7 @@ A Multiverse has the following structure:
 * Upper tree structure:
 ** <code>key</code>: <code>asset_id</code> or <code>sha256(asset_key_family)</code>
-** <code>value</code>: <code>asset_group_tree_root || asset_group_sum</code>
+** <code>value</code>: <code>inner_tree_root</code>
```
Maybe keep `asset_group_tree_root` as adds more context? Then can _just_ drop the sum value there in the key since that's implicit.

Roasbeef reviewed (approved):
LGTM ⛄️
